src/services/EdFi.DataManagementService.Api/Core/Pipeline/PipelineProvider.cs

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+namespace EdFi.DataManagementService.Core.Pipeline;
+
+/// <summary>
+/// Implements a simple Data Management Service pipeline provider, which
+/// is modeled on the Middleware pattern.
+///
+/// Add pipeline steps in execution order with StartWith() and AndThen().
+/// Run the pipeline steps with Run().
+/// </summary>
+public class PipelineProvider
+{
+    private readonly List<IPipelineStep> _steps = [];
```

Comment:
Wondering if we have any potential for thread safety problems with this approach. There's a performance hit to using a locking collection, so we don't want to overdo it. Haven't fully thought this through yet... probably fine since the pipeline is setup and then used, but not altered.

https://learn.microsoft.com/en-us/dotnet/standard/collections/thread-safe/

Reply:
The `AndThen()` stuff was cool, but because it was stateful issue I dropped in in favor of just a constructor with the whole list. Wrapped construction in the facade with a `Lazy<>`  and we are good to go.